src/featbin/wav-reverberate.cc

```diff
@@ -0,0 +1,250 @@
+// featbin/extract-segments.cc
+
+// Copyright 2015  Tom Ko
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "feat/wave-reader.h"
+
+namespace kaldi {
+
+void FFTConvolution(Vector<BaseFloat> &x, Vector<BaseFloat> &h) {
+  int xdim = x.Dim();
+  int M = h.Dim();
+
+  KALDI_LOG << "M is " << M;
+
+  int N = 2;
```

Comment:
Actually, that looks like a temporary logging statement that should be
removed entirely (programs should not produce spam on the output).
Also, the name is not Google compliant, should be FftConvolution, but also
function names should be verbs not nouns so it needs further change.  [I'm
not sure what it does so I can't say].  And non-const references as
function arguments are not allowed.

Dan


On Mon, Aug 17, 2015 at 1:09 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In src/featbin/wav-reverberate.cc
> <https://github.com/kaldi-asr/kaldi/pull/76#discussion_r37230847>:
>
> > +// See the Apache 2 License for the specific language governing permissions and
> > +// limitations under the License.
> > +
> > +#include "base/kaldi-common.h"
> > +#include "util/common-utils.h"
> > +#include "feat/wave-reader.h"
> > +
> > +namespace kaldi {
> > +
> > +void FFTConvolution(Vector<BaseFloat> &x, Vector<BaseFloat> &h) {
> > +  int xdim = x.Dim();
> > +  int M = h.Dim();
> > +
> > +  KALDI_LOG << "M is " << M;
> > +
> > +  int N = 2;
>
> use more descriptive names for the variables. e.g. signal_length,
> filter_length, output_length.
>
> If you are using the variables names to reflect corresponding quantities
> in an equation it is advisable to have the equations in the comments, to
> ensure that the code is easy to follow.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/76/files#r37230847>.
>